app/javascript/components/supplemental_application/sections/Lease.js

```diff
@@ -126,18 +127,45 @@ const Lease = ({ form, values, store }) => {
     ...map(confirmedPreferences, pluck('id', 'preference_name'))
   ])
 
-  const getVisited = (fieldName) => form.getFieldState(fieldName)?.visited
+  const availableUnits = units.filter(
+    (unit) => !unit.application_id || unit.application_id === application.id
+  )
 
-  const openDeleteLeaseConfirmation = () => setShowDeleteConfirmation(true)
-  const closeDeleteLeaseConfirmation = () => setShowDeleteConfirmation(false)
+  const availableUnitsOptions = formUtils.toOptions(
+    map(availableUnits, pluck('id', 'unit_number', 'priority_type'))
+  )
 
-  const handleDeleteLeaseConfirmed = () => {
-    setShowDeleteConfirmation(false)
-    handleDeleteLease()
-  }
+  const selectedUnit = form.getState().values.lease?.unit
 
   const areNoUnitsAvailable = !availableUnitsOptions.length
 
+  const prefUsedIs = (pref) => {
```

Comment:
nit: prefUsedIs as a name is confusing to me. Is this saying "The preference selected on the current application is using `pref`"? If so I think an even simpler name like `isSelected(pref)` works.